packages/amplify-graphql-schema-generator/src/ts-schema-generator/ts-schema-generator.ts

```diff
@@ -0,0 +1,52 @@
+import { DataSourceAdapter, MySQLDataSourceAdapter, PostgresDataSourceAdapter } from '../datasource-adapter';
+import { DBEngineType, Engine, Schema } from '../schema-representation';
+import { generateTypescriptDataSchema } from './generate-ts-schema';
+
+// This is the contract for the customer facing API to provide the database configuration in a typescript file.
+export type TypescriptDataSchemaGeneratorConfig = {
+  engine: 'mysql' | 'postgresql';
```

Comment:
I've considered using DBEngineType here. However for the customer facing API, we need this in all lowercase.